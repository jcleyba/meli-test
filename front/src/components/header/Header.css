.meli-header {
  background: #ffe600;
  display: flex;
  justify-content: center;
  align-items: center; }
  .meli-header .logo {
    max-width: 70px;
    margin: 5px 20px 5px 0; }
    .meli-header .logo img {
      width: 100%; }
