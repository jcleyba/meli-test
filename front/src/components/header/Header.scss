@import '../../styles/_variables';

.meli-header {
  background: $meli_yellow;
  display: flex;
  justify-content: center;
  align-items: center;

  .logo {
    max-width: 70px;
    margin: 5px 20px 5px 0;

    img {
      width: 100%;
    }
  }
}
