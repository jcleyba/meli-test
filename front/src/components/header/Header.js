import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from 'react-router-dom';
import queryString from 'query-string';

import { fetchItems } from '../../actions/searchActions'

import SearchBar from '../common/SearchBar';

import logo from '../../images/Logo_ML@2x.png.png';
import './Header.css';

const ITEMS_TO_FETCH = 4;

export class Header extends Component {

  getTerm = () => {
    const term = queryString.parse(this.props.location.search);

    return term.search || '';
  };

  searchHandler = (term) => {
    this.props.history.push(`/items?search=${term}`);
    this.props.fetchItems(term, ITEMS_TO_FETCH)
  };

  render() {
    return (
      <header className="meli-header">
        <div className="logo">
          <NavLink to="/">
            <img src={logo} alt="logo" title="logo"/>
          </NavLink>
        </div>
        <SearchBar term={this.getTerm()}
                   onSubmit={this.searchHandler}/>
      </header>
    )
  }
};

export default withRouter(connect(null, {fetchItems})(Header));
