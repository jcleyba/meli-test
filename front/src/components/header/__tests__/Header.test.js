import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-enzyme';
import { MemoryRouter } from 'react-router-dom';

import { Header } from '../Header';

describe('Header Component', () => {

  let wrapper;
  let mounted;

  const props = {
    fetchItems: jest.fn(),
    location: {
      search: ''
    }
  };

  const Component = (props) => {
    return <MemoryRouter>
      <Header {...props}/>
    </MemoryRouter>
  };


  beforeEach(() => {
    wrapper = shallow(<Component {...props}/>);
    mounted = mount(<Component {...props}/>);
  });

  afterEach(() => {
    wrapper = null;
    mounted = null;
  });

  describe('Header', () => {

    it('matches current snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have fetchItems prop', () => {
      expect(mounted).toHaveProp('fetchItems');
    });

    it('should have a logo present', () => {
      expect(mounted.find('.logo')).toBePresent();
    });

    it('should contain a SearchBar', () => {
      expect(mounted.find('SearchBar')).toBePresent();
    });
  })

});
