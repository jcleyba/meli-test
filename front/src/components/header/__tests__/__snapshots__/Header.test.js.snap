// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Header matches current snapshot 1`] = `"<header class=\\"meli-header\\"><div class=\\"logo\\"><a class=\\"active\\" aria-current=\\"true\\" href=\\"/\\"><img src=\\"Logo_ML@2x.png.png\\" alt=\\"logo\\" title=\\"logo\\"/></a></div><div class=\\"search-bar\\"><form><input type=\\"text\\" name=\\"search\\" class=\\"search-input\\" placeholder=\\"Nunca dejes de buscar\\" value=\\"\\"/><button type=\\"submit\\" class=\\"search-btn\\"></button></form></div></header>"`;
